components/kyma-environment-broker/common/gardener/client.go

```diff
@@ -4,56 +4,68 @@ import (
 	"fmt"
 	"io/ioutil"
 
-	gardener_apis "github.com/gardener/gardener/pkg/client/core/clientset/versioned/typed/core/v1beta1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
 	restclient "k8s.io/client-go/rest"
 	"k8s.io/client-go/tools/clientcmd"
 )
 
-func NewGardenerClusterConfig(kubeconfigPath string) (*restclient.Config, error) {
+type SecretBinding struct {
+	unstructured.Unstructured
+}
 
-	rawKubeconfig, err := ioutil.ReadFile(kubeconfigPath)
-	if err != nil {
-		return nil, fmt.Errorf("failed to read Gardener Kubeconfig from path %s: %s", kubeconfigPath, err.Error())
-	}
+func (b SecretBinding) GetSecretRefName() string {
+	//TODO: implement jw check errors
+	str, _, _ := unstructured.NestedString(b.Unstructured, "spec", "secretRef", "name")
+	return str
+}
 
-	gardenerClusterConfig, err := RESTConfig(rawKubeconfig)
-	if err != nil {
-		return nil, fmt.Errorf("")
-	}
+type Shoot struct {
+	unstructured.Unstructured
+}
 
-	return gardenerClusterConfig, nil
+func (b Shoot) GetSpecSecretBindingName() string {
+	//TODO: implement jw check errors
+	str, _, _ := unstructured.NestedString(b.Unstructured, "spec", "secretBinding", "name")
+	return str
 }
 
-func NewGardenerSecretBindingsInterface(gardenerClient *gardener_apis.CoreV1beta1Client, gardenerProjectName string) gardener_apis.SecretBindingInterface {
-	gardenerNamespace := gardenerNamespace(gardenerProjectName)
-	return gardenerClient.SecretBindings(gardenerNamespace)
+func (b Shoot) GetSpecMaintenanceTimeWindowBegin() string {
+	//TODO: implement jw check errors
+	str, _, _ := unstructured.NestedString(b.Unstructured, "spec", "maintenance", "timeWindow", "begin")
+	return str
 }
 
-func NewGardenerShootInterface(gardenerClusterCfg *restclient.Config, gardenerProjectName string) (gardener_apis.ShootInterface, error) {
+func (b Shoot) GetSpecMaintenanceTimeWindowEnd() string {
+	//TODO: implement jw check errors
+	str, _, _ := unstructured.NestedString(b.Unstructured, "spec", "maintenance", "timeWindow", "end")
+	return str
+}
 
-	gardenerNamespace := gardenerNamespace(gardenerProjectName)
+func (b Shoot) GetSpecRegion() string {
+	//TODO: implement jw check errors
+	str, _, _ := unstructured.NestedString(b.Unstructured, "spec", "region")
+	return str
+}
 
-	gardenerClusterClient, err := gardener_apis.NewForConfig(gardenerClusterCfg)
-	if err != nil {
-		return nil, err
-	}
+var SecretBindingResource = schema.GroupVersionResource{Group: "core.gardener.cloud", Version: "v1beta1", Resource: "secretbindings"}
```

Comment:
@szwedm this would be GroupVersionResource that needs to be passed to the dynamic client `.Resource(gvr)`